subsys/nrf_security/src/psa_crypto_driver_wrappers.c

```diff
@@ -2413,44 +2421,40 @@ psa_status_t psa_driver_wrapper_asymmetric_decrypt(
 
 psa_status_t psa_driver_wrapper_init_random(psa_driver_random_context_t *context)
 {
-#if defined(PSA_CRYPTO_DRIVER_ALG_PRNG_CC3XX_PLATFORM)
-	/* Using internal context. */
-	(void)context;
 
-	return PSA_SUCCESS;
-#endif
-
-#if defined(PSA_CRYPTO_DRIVER_ALG_PRNG_OBERON)
-#if defined(PSA_CRYPTO_DRIVER_ALG_CTR_DRBG_OBERON)
+#if defined(PSA_NEED_OBERON_CTR_DRBG_DRIVER)
 	return oberon_ctr_drbg_init(&context->oberon_ctr_drbg_ctx);
-#elif defined(PSA_CRYPTO_DRIVER_ALG_HMAC_DRBG_OBERON)
+#elif defined(PSA_NEED_OBERON_HMAC_DRBG_DRIVER)
 	return oberon_hmac_drbg_init(&context->oberon_hmac_drbg_ctx);
-#endif
-#endif /* defined(PSA_CRYPTO_DRIVER_ALG_PRNG_OBERON) */
-
+#else
+	/*
+	 * When PSA_NEED_CC3XX_CTR_DRBG_DRIVER or PSA_NEED_CC3XX_HMAC_DRBG_DRIVER or test DRBG
+	 * is enabled no initialization is required because an internal context is used.
```

Comment:
It's also possible to disable PSA_WANT_GENERATE_RANDOM, and in that case, "init_random" should also return sucsses, but "generate_random" returns failure. This comment does not include this case.

Reply:
Maybe I can append something like this:
"When PSA_WANT_GENERATE_RANDOM is not enabled this will return success, because the
  psa_crypto_init calls this function." 
  
  ?